### ISSUE TITLE
auditing code base using Coverity

### ISSUE THREAD

**Issue by landroni** [OtterBrowser/otter-browser#693]
Have you considered auditing your code using the automated scan utility Coverity? 
https://scan.coverity.com/

Their tool is being used by heavy-hitters like LibreOffice, Linux Kernel, Firefox or Python. It may be worth submitting encryptr to Coverity, too, to watch out for pesky implementation omissions. 

If there is interest, I could set up the coverity scan and perform an initial assessment of defects.

**Comment by Emdek** [comment]
It seems to be a good idea. :-)

**Comment by srgloureiro** [comment]
By the way. some staff at Codegears decided to anlyze the Unreal Engine code using Cppdepnd and CQLinq, described in the article http://www.codergears.com/Blog/?p=965

In my opinion, it is needed to know first the best practices, and the link speaks about some ones in C++.

**Comment by The-Compiler** [comment]
Some days ago I also set up a buildbot for otter which is already running:

http://qutebrowser.org:8011/waterfall

It currently runs `cppcheck` and `scan-build` (from clang).

**Comment by landroni** [comment]
Here we go: 

https://scan.coverity.com/projects/4221

The code is pretty uniformly clean, with no high impact issues, and with only 0.12 Defect Density. You need to connect with your Github account to view the bugs.

**Comment by uahim** [comment]
excuse this question from a technically interested casual user: 164,617 lines of code? is that all written exclusively for Otter or does it also include QtWebKit?

**Comment by landroni** [comment]
The stats concern only Otter, so I suspect it's  164,617 lines of C code in Otter.

**Comment by The-Compiler** [comment]
Hm, I wonder where it's getting this metric from - I get something quite different:

```
$ find . \( -iname '*.cpp' -o -iname '*.h' -o -iname '*.js' \) -exec cat {} \; | wc -l  # inside the src/ dir
41839
```

**Comment by landroni** [comment]
Yeah, not sure about that. Maybe they explain their methodology in the docs...

**Comment by landroni** [comment]
As per http://codemonkey.org.uk/2014/08/13/year-coverity-linux-kernel-scans/

> (LOC based on C preprocessor output)

I don't know what that means...

**Comment by ShadowCreator** [comment]
@landroni That probably means the lines of code count includes the expanded #include sections in the code.

**Comment by Zero3K** [comment]
It should be ran through it again in order to see if any new bugs have popped up.
